portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -1,143 +1,155 @@
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
 package com.google.sps.servlets;
 
-import java.io.IOException;
-import javax.servlet.annotation.WebServlet;
-import javax.servlet.http.HttpServlet;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.users.User; 
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
 import com.google.appengine.api.datastore.DatastoreService;
-import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
-import com.google.appengine.api.datastore.FetchOptions; 
-import com.google.appengine.api.datastore.FetchOptions.Builder; 
-import com.google.gson.Gson;
+import com.google.appengine.api.datastore.FetchOptions;
 import com.google.sps.data.Quote; 
+import com.google.sps.data.Util; 
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
 import java.util.List; 
-import java.util.ArrayList; 
+import java.util.ArrayList;
+import java.util.Map; 
+import java.util.HashMap; 
+import java.util.Optional;
 
-/** Servlet that returns some example content. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private static final String QUOTE = "Quote";
-  private static final String TEXT = "text";
-  private static final String TIMESTAMP = "timestamp"; 
+  private static final String INDEX_URL = "/index.html"; 
+
+  // Constants for property of a Quote item in Datastore.
+  private static final String PROPERTY_NAME_QUOTE = "Quote";
+  private static final String PROPERTY_NAME_TEXT = "text";
+  private static final String PROPERTY_NAME_TIMESTAMP = "timestamp"; 
+  private static final String PROPERTY_NAME_USER_EMAIL = "userEmail";
+  private static final String PROPERTY_NAME_NICKNAME = "nickname";
+
+  // Constants for information to be put into response of a GET request.
+  private static final String IS_USER_LOGGED_IN = "loggedIn"; 
+  private static final String REDIRECT_URL_PARAM = "redirectUrl";
+  private static final String HOME_URL = "/"; 
+
+  // Constant for fetching quotes from Datastore
   private static final int DEFAULT_NUM_QUOTES = 5;
 
-  /**
-    * Retrives all quotes that are stored in Datastore
-    * and puts them into a list of quotes. This list is then written
-    * as the response from the server. 
-    */
+  // Global variable for an instance of the Util helper class
+  private static Util util; 
```

Comment:
Once you make the `getUserNickname(...)` method static you don't need an instance of `Util` here because you can just call the static methods directly like `Util.getDatastoreServiceWithConsistency()`.